docs/proposals/26-workload-identity.md

```diff
@@ -0,0 +1,514 @@
+service, hence no static credentials like service keys are used. More details
+for the cloud providers, can be found at their documentation:
+
+- [AWS: Creating OpenID Connect (OIDC) identity providers](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html)
+- [Azure: Workload identity federation](https://learn.microsoft.com/en-us/entra/workload-id/workload-identity-federation)
+- [Alicloud: Overview of OIDC-based SSO](https://www.alibabacloud.com/help/en/ram/user-guide/overview-of-oidc-based-sso)
+- [GCP: Workload identity federation](https://cloud.google.com/iam/docs/workload-identity-federation)
+
+## Motivation
+
+Gardener is using variety of external services to create different resources
+needed for the lifecycle of a shoot or seed cluster, resources like virtual
+machines, volumes, object storage, load balancers, DNS records, etc. Each
+request to create, update or delete such resource needs to be authenticated with
+some kind of credentials. Sometimes the resources reside in different cloud
+accounts and credentials with mixed ownership are used, for example when
+Gardener Operator and Gardener User are different entities, Gardener Users
+brings their own credentials for their account and let Gardener use them on
+their behalf to create resources. These credentials usually have long lifetime
+(and are often non-expiring), they are reused in different scenarios by various
+tools, occasionally granted with broader permissions, stored in different
+locations. Such handling poses various security risks.
+
+The static long-lived credentials can be replaced with short-lived auto-rotated
+JWTs issued and used only by Gardener, never leaving the Gardener environment
+except for representing Gardener workloads before an external entity,
+eliminating the security burden to manage and store static credentials. Static
+credentials can expire or get accidentally invalidated, which will cause
+reconciliation flows to fail preventing delivery of updates, fixes, and
+improvements. This risk is better to be managed in an automated way by Gardener
+itself.
+
+The JSON Web Tokens are ephemeral and not stored anywhere by the issuer, a
+feature that Gardener can benefit of as well because it will not have to store
+the credentials of Gardener users.
+
+### Goals
+
+- Manage shoot clusters without credentials provided by the user.
+- Replace static credentials for gardener system components, e.g. DNS and backup
+  controllers.
+- Rotate credentials regularly. Rotating the token signing keys, will
+  effectively invalidate all previously issued tokens.
+- Offload Gardener users from the burden to store and manage static credentials
+  for their accounts.
+
+### Non-Goals
+
+- Register gardener as trusted identity in the shoot clusters.
+- The tokens to be usable for authentication with the Gardener API.
+- Compatibility with gardenctl integration with cloud provider CLIs.
```

Comment:
If this is a non-goal, how do you envision this to work in the future? cc @petersutter

Reply:
why wouldn't gardenctl not directly read the secret? 

Currently our endusers can use the [`provider-env`](https://github.com/gardener/gardenctl-v2/blob/master/docs/help/gardenctl_provider-env.md) command to configure their cloud provider CLIs. With this approach, endusers would not be able to do so as they do not have access to the control plane.

Why not directly create the `security.gardener.cloud/v1alpha1.TokenRequest` from the CLI?

Reply:
then, regardless of the enduser access to the cloud account, how do you envision operators troubleshooting issues in the future without what you called "escalating privileges", where they previously used the shoot infra credentials